moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py

```diff
@@ -79,14 +86,21 @@ def generate_launch_description():
                      output='log',
                      arguments=['0.0', '0.0', '0.0', '0.0', '0.0', '0.0', 'world', 'panda_link0'])
 
-    # Fake joint driver
-    fake_joint_driver_node = Node(package='fake_joint_driver',
-                                  node_executable='fake_joint_driver_node',
-                                  # TODO(JafarAbdi): Why this launch the two nodes (controller manager and the fake joint driver) with the same name!
-                                  # node_name='fake_joint_driver_node',
-                                  parameters=[os.path.join(get_package_share_directory("run_moveit_cpp"), "config", "panda_controllers.yaml"),
-                                              os.path.join(get_package_share_directory("run_moveit_cpp"), "config", "start_positions.yaml"),
-                                              robot_description]
-                                  )
-
-    return LaunchDescription([ static_tf, rviz_node, run_moveit_cpp_node, fake_joint_driver_node ])
+    # Joint state publisher
+    joint_state_publisher = Node(package='joint_state_publisher',
+                                 node_executable='joint_state_publisher',
+                                 node_name='joint_state_publisher',
+                                 arguments=[os.path.join(get_package_share_directory('moveit_resources'), 'panda_description/urdf/panda.urdf')],
+                                 output='log',
+                                 parameters=[{'source_list': ['/fake_controller_joint_states'],
+                                              'zeros': {'panda_joint1': 0.0,
```

Comment:
Can't we use the initial pose instead of 'zeros'?

Reply:
joint_state_publisher keep publishing zero values (the robot is in collision with these joint state) until it gets a message from MoveIt fake controller which takes some time to initialize causing MoveItCpp to get the wrong angles

Reply:
Setting `publish_default_positions=false` should do the trick